cache/network/download.go

```diff
@@ -48,12 +50,40 @@ func Download(ctx context.Context, params DownloadParams, logger log.Logger) (ma
 
 	logger.Debugf("Download archive")
 
-	downloadErr := downloadFile(ctx, retryableHTTPClient.StandardClient(), restoreResponse.URL, params.DownloadPath)
-	if downloadErr != nil {
-		return "", fmt.Errorf("failed to download archive: %w", downloadErr)
+	const maxRetries = 3
+	const retryDelay = 2 * time.Second
+	retriableErrors := []string{"Range request returned invalid Content-Length", "EOF"}
+
+	// Attempt to download, retrying on retriable errors
+	for attempt := 0; attempt < maxRetries; attempt++ {
+		downloadErr := downloadFile(ctx, retryableHTTPClient.StandardClient(), restoreResponse.URL, params.DownloadPath)
```

Comment:
We should check what happens if the target file already exists - and ensure that won't cause an issue (delete it if there's already a partially downloaded file if that'd cause an issue).

Reply:
I checked&tested, download is an idempotent operation
https://github.com/melbahja/got/blob/9c99581287dd94c9fceee95e5c9b502941903497/download.go#L201